hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileRequest.java

```diff
@@ -84,6 +92,70 @@ public static OMDirectoryResult verifyFilesInPath(
     return OMDirectoryResult.NONE;
   }
 
+  /**
+   * generate the object id from the transaction id.
+   * @param id
+   * @return object id
+   */
+  public static long getObjIdFromTxId(long id) {
+    return id << TRANSACTION_ID_SHIFT;
+  }
+
+  /**
+   * Return list of missing parent directories in the given path.
+   * @param omMetadataManager
+   * @param volumeName
+   * @param bucketName
+   * @param keyPath
+   * @return List of keys representing non-existent parent dirs
+   * @throws IOException
+   */
+  public static List<String> getMissingParents(
+      @Nonnull OMMetadataManager omMetadataManager,
+      @Nonnull String volumeName,
+      @Nonnull String bucketName,
+      @Nonnull Path keyPath) throws IOException {
+
+    List<String> missing = new ArrayList<>();
+
+    while (keyPath != null) {
+      String pathName = keyPath.toString();
+
+      String dbKeyName = omMetadataManager.getOzoneKey(volumeName,
+          bucketName, pathName);
+      String dbDirKeyName = omMetadataManager.getOzoneDirKey(volumeName,
+          bucketName, pathName);
+
+      if (omMetadataManager.getKeyTable().isExist(dbKeyName)) {
+        // Found a file in the given path.
+        String errorMsg = "File " + dbKeyName + " exists with same name as " +
+            " directory in path : " + pathName;
+        throw new IOException(errorMsg);
+      } else if (omMetadataManager.getKeyTable().isExist(dbDirKeyName)) {
+        // Found a directory in the given path. Higher parents must exist.
+        break;
+      } else {
+        missing.add(pathName);
+      }
+      keyPath = keyPath.getParent();
+    }
+
+    return missing;
+  }
+
+  private static OmKeyInfo getKeyInfo(
```

Comment:
There seems to be no usage for this method.

Reply:
thanks for pointing it out, will remove. a relic from an earlier version of my patch.